src/movies.js

```diff
@@ -1,25 +1,84 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
-// How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
 
-// Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
 
-// Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
 
-// Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
 
-// BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes(moviesArray) {}
 
-// BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg(moviesArray) {}
+
+// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+// How could you "clean" a bit this array and make it unified (without duplicates)?
+function getAllDirectors(moviesArray) {
+    const newArr = moviesArray.map((movie) => movie.director)  
+    return newArr;
+    };
+    getAllDirectors(movies)
+    
+    // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+     function howManyMovies(moviesArray) {
+     const newArray= moviesArray.filter((movie) => 
+     movie.director === "Steven Spielberg" && movie.genre.includes("Drama"))
+     return newArray.length;
+    };
+    
+    // Iteration 3: All scores average - Get the average of all scores with 2 decimals
+    function scoresAverage(moviesArray) {
+        if (moviesArray.length === 0) return 0;
```

Comment:
There's nothing wrong here, however, the reduce function won't execute if the moviesArray is empty. So, the code block `if (!moviesArray.length) return 0;` is not needed as the function will work fine without it, it will return 0 as default value for empty array.